### PR TITLE
support validate setex headings

### PR DESCRIPTION
**Pull request by OsmondJiang**
setex headings is not allowed in docs repository

[AB#211394](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/211394)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5848)

### CODE REVIEW

File: docs/specs/validation.yml [hunk 1]
```diff
@@ -733,4 +733,67 @@ outputs:
...
+        ------
+
+    [!include[](b.md)]
+  b.md: |
+    Title 2
+    ------
+outputs:
+  a.json:
+  .errors.log: |
+    ["suggestion","headings-with-underline","Bad 1 format: 'Title 1'. Use hashes to create H1(#) or H2(##)","a.md",5,1]
```
Comment (yufeih) [inline docs/specs/validation.yml:796]: Is it `Bad h1 format`? `Bad 1 format` seems weird.

File: src/docfx/lib/markdown/validation/ContentValidationExtension.cs [hunk 2]
```diff
@@ -33,9 +33,8 @@ internal static class ContentValidationExtension
                                 Level = headingBlock.Level,
                                 SourceInfo = headingBlock.ToSourceInfo(),
                                 Content = GetHeadingContent(headingBlock), // used for reporting
-
-                                // HeadingChar = headingBlock.HeadingChar
-                                // RenderedPlainText = MarkdigUtility.ToPlainText(headingBlock), // used for validation
+                                HeadingChar = headingBlock.HeaderChar,
+                                RenderedPlainText = MarkdigUtility.ToPlainText(headingBlock), // used for validation
```
Comment (yufeih) [inline src/docfx/lib/markdown/validation/ContentValidationExtension.cs:37]: What's the difference between `Content` and `RenderedPlainText`?

File: docs/specs/validation.yml (hunk 1 above)
Comment (OsmondJiang) [reply docs/specs/validation.yml:796]: it's generated from rule, we can adjust the rule at any time.

File: src/docfx/lib/markdown/validation/ContentValidationExtension.cs (hunk 2 above)
Comment (OsmondJiang) [reply src/docfx/lib/markdown/validation/ContentValidationExtension.cs:37]: offline talked, content is the user input, rendered plain text is the build output. the previous one is for reporting, the later one is for validation.

**Pull request merged**